src/Time/StepChoosers/Factory.hpp

```diff
@@ -0,0 +1,58 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "Time/StepChoosers/Cfl.hpp"
+#include "Time/StepChoosers/Constant.hpp"
+#include "Time/StepChoosers/Increase.hpp"
+#include "Time/StepChoosers/PreventRapidIncrease.hpp"
+#include "Time/StepChoosers/StepToTimes.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace Frame {
+struct Inertial;
+}  // namespace Frame
+/// \endcond
+
+namespace StepChoosers {
+namespace Factory_detail {
+template <typename Use, size_t VolumeDim, typename System,
+          bool HasCharSpeedFunctions>
+using common_step_choosers = tmpl::push_back<
+    tmpl::conditional_t<
+        HasCharSpeedFunctions,
+        tmpl::list<StepChoosers::Cfl<Use, VolumeDim, Frame::Inertial, System>>,
```

Comment:
You can get the volume_dim from the system, you could remove the VolumeDim template parameter from a lot of these places if you want.

Reply:
Added a commit at the start that removes the parameter from `Cfl` so these lists no longer need it at all.